pkg/gardenlet/controller/networkpolicy/reconciler.go

```diff
@@ -391,11 +393,13 @@ func (r *Reconciler) reconcileNetworkPolicyAllowToShootNetworks(ctx context.Cont
 	}
 
 	var shootNetworks []string
-	if v := shoot.Spec.Networking.Nodes; v != nil {
-		shootNetworks = append(shootNetworks, *v)
-	}
-	if v := shoot.Spec.Networking.Pods; v != nil {
-		shootNetworks = append(shootNetworks, *v)
+	if !shoot.IsWorkerless() {
+		if v := shoot.Spec.Networking.Nodes; v != nil {
+			shootNetworks = append(shootNetworks, *v)
+		}
+		if v := shoot.Spec.Networking.Pods; v != nil {
+			shootNetworks = append(shootNetworks, *v)
+		}
 	}
 	if v := shoot.Spec.Networking.Services; v != nil {
```

Comment:
Same here

Reply:
I added the check in the predicate, now the field cannot be nil inside the reconciler.